NitroxClient/Communication/Abstract/DeferredPacketReceiver.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.Packets;
+
+namespace NitroxClient.Communication.Abstract
+{
+    public abstract class DeferredPacketReceiver
```

Comment:
Shouldn't the methods in this class be marked `abstract` as well, to enforce them being implemented by the subclass?

I've been out of the loop for too long, but these "default implementations" seem wrong to me.

Reply:
Also, what is the use-case for having multiple implementations of a DeferredPacketReceiver?  I am only seeing one in the PR but may be missing something.